backend/src/main/java/com/allog/dallog/domain/categoryrole/exception/CategoryRoleConcurrencyException.java

```diff
@@ -0,0 +1,8 @@
+package com.allog.dallog.domain.categoryrole.exception;
+
+public class CategoryRoleConcurrencyException extends RuntimeException {
+
+    public CategoryRoleConcurrencyException() {
+        super("누군가 회원님의 카테고리 역할을 수정했습니다. 다시 시도해주세요.");
+    }
```

Comment:
꼭 동시성 예외를 따로 만들어야 할까요? 기술 문제보다는 비즈니스 기준으로 예외를 다루는 게 좋을 것 같아요!
기존의 예외를 재사용하고 메시지만 바꿔줄 수 있을 것 같아요.
```suggestion
    public NotAbleToChangeRoleException() {
        super("회원님의 권한이 변경되어 카테고리 역할을 수정할 수 없습니다.");
    }
```

Reply:
```java
} catch (final ObjectOptimisticLockingFailureException e) {
    throw NotAbleToChangeRoleException.concurrentIssue();
}
```

이렇게 기존 예외를 재사용했습니다! 정적 팩토리 메소드를 사용해서 별도로 메시지를 파라미터로 전송하지 않아도 되도록 했어요.